File: app/src/main/java/com/cdnbye/p2pengine/MenuActivity.java
package com.cdnbye.p2pengine;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListAdapter;
import android.widget.ListView;
import androidx.fragment.app.FragmentActivity;
import java.util.Arrays;
import java.util.List;

import com.cdnbye.core.p2p.EngineExceptionListener;
import com.cdnbye.core.utils.EngineException;
import com.cdnbye.core.utils.LogLevel;
import com.cdnbye.core.p2p.P2pConfig;
import com.cdnbye.sdk.P2pEngine;

public class MenuActivity extends FragmentActivity {

    ListView listView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        listView = findViewById(R.id.listView);
        ListAdapter adapter = new ArrayAdapter<>(this, android.R.layout.simple_list_item_1, android.R.id.text1, buildListData());
        listView.setAdapter(adapter);

        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> adapterView, View view, int position, long l) {
                ListEntry item = (ListEntry) listView.getAdapter().getItem(position);
                startActivity(new Intent(MenuActivity.this, item.activityClass));
            }
        });

        P2pConfig config = new P2pConfig.Builder()
                .logEnabled(true)
                .logLevel(LogLevel.DEBUG)
                .p2pEnabled(true)
                .waitForPeer(true)
                .build();

        P2pEngine.init(getApplicationContext(), "ZMuO5qHZg", config);

        P2pEngine.getInstance().registerExceptionListener(new EngineExceptionListener() {
            @Override
            public void onTrackerException(EngineException e) {
                System.out.println("onTrackerException " + e.getMessage());
            }

            @Override
            public void onSignalException(EngineException e) {
                System.out.println("onSignalException " + e.getMessage());
            }

            @Override
            public void onSchedulerException(EngineException e) {
                System.out.println("onSchedulerException " + e.getMessage());
            }

            @Override
            public void onOtherException(EngineException e) {
                System.out.println("onOtherException " + e.getMessage());
            }
        });
    }

    private List<ListEntry> buildListData() {
        return Arrays.asList(
                new ListEntry("Video Stream P2P", PlayerActivity.class),
                new ListEntry("File Download P2P", DownloadActivity.class)
        );
    }

    private static final class ListEntry {

        private final String title;
        private final Class activityClass;

        public ListEntry(String title, Class activityClass) {
            this.title = title;
            this.activityClass = activityClass;
        }

        @Override
        public String toString() {
            return title;
        }
    }

}

File: app/src/main/java/com/cdnbye/demomedia3/MenuActivity.kt
package com.cdnbye.demomedia3

import android.R
import android.content.Intent
import android.os.Bundle
import android.support.multidex.MultiDex
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.ListAdapter
import android.widget.ListView
import androidx.fragment.app.FragmentActivity
import com.cdnbye.demomedia3.databinding.ActivityMenuBinding
import com.p2pengine.core.p2p.EngineExceptionListener
import com.p2pengine.core.p2p.P2pConfig
import com.p2pengine.core.tracking.TrackerZone
import com.p2pengine.core.utils.EngineException
import com.p2pengine.core.utils.LogLevel
import com.p2pengine.sdk.P2pEngine
import java.util.*

class MenuActivity : FragmentActivity() {
    private lateinit var menuBinding: ActivityMenuBinding
    private lateinit var listView: ListView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        MultiDex.install(this)
        menuBinding = ActivityMenuBinding.inflate(layoutInflater)

        setContentView(menuBinding.root)

        listView = menuBinding.listView
        val adapter: ListAdapter = ArrayAdapter(
            this,
            R.layout.simple_list_item_1,
            R.id.text1,
            buildListData()
        )
        listView.adapter = adapter
        listView.onItemClickListener =
            AdapterView.OnItemClickListener { _, _, position, _ ->
                val item: ListEntry =
                    listView.adapter.getItem(position) as ListEntry
                startActivity(Intent(this@MenuActivity, item.activityClass))
        }

        val config = P2pConfig.Builder()
            .p2pEnabled(true)
            .logEnabled(true)
            .logLevel(LogLevel.DEBUG)
            .trackerZone(TrackerZone.Europe)
//            .trackerZone(TrackerZone.HongKong)
//            .trackerZone(TrackerZone.USA)
            .build()

        println("MainActivity P2pEngine init")
        P2pEngine.init(applicationContext, "ZMuO5qHZg", config)

        P2pEngine.instance?.registerExceptionListener(object : EngineExceptionListener {
            override fun onTrackerException(e: EngineException) {
                println("onTrackerException ${e.message}")
            }

            override fun onSignalException(e: EngineException) {
                println("onSignalException ${e.message}")
            }

            override fun onSchedulerException(e: EngineException) {
                println("onSchedulerException ${e.message}")
            }

            override fun onOtherException(e: EngineException) {
                println("onOtherException ${e.message}")
            }

        })

    }

    private fun buildListData(): List<ListEntry> {
        return Arrays.asList(
            ListEntry("EXO", ExoActivity::class.java),
        )
    }

    private class ListEntry(private val title: String, val activityClass: Class<*>) {
        override fun toString(): String {
            return title
        }
    }


}